piIOComm.c

```diff
@@ -319,83 +319,6 @@ INT32S piIoComm_sendRS485Tel(INT16U i16uCmd_p, INT8U i8uAddress_p,
 	return 0;
 }
 
-INT32S piIoComm_sendTelegram(SIOGeneric * pRequest_p, SIOGeneric * pResponse_p)
-{
-	INT32S i32uRv_l = 0;
-	INT8U len_l;
-	int ret;
-#if 0				//def DEBUG_DEVICE_IO
-	static INT8U last_out[40][2];
-	static INT8U last_in[40][2];
-#endif
-
-	len_l = pRequest_p->uHeader.sHeaderTyp1.bitLength;
-
-	pRequest_p->ai8uData[len_l] = piIoComm_Crc8((INT8U *) pRequest_p, IOPROTOCOL_HEADER_LENGTH + len_l);
-
-#if 0				//def DEBUG_DEVICE_IO
-	if (last_out[pRequest_p->uHeader.sHeaderTyp1.bitAddress][0] != pRequest_p->ai8uData[0]
-	    || last_out[pRequest_p->uHeader.sHeaderTyp1.bitAddress][1] != pRequest_p->ai8uData[1]) {
-		last_out[pRequest_p->uHeader.sHeaderTyp1.bitAddress][0] = pRequest_p->ai8uData[0];
-		last_out[pRequest_p->uHeader.sHeaderTyp1.bitAddress][1] = pRequest_p->ai8uData[1];
-		pr_info("dev %2d: send cyclic Data addr %d + %d output 0x%02x 0x%02x\n",
-			pRequest_p->uHeader.sHeaderTyp1.bitAddress,
-			pRequest_p->uHeader.sHeaderTyp1.bitAddress,
-			RevPiDevice.dev[i8uDevice_p].i16uOutputOffset, sRequest_l.ai8uData[0], sRequest_l.ai8uData[1]);
-	}
-#endif
-
-	ret = piIoComm_send((INT8U *) pRequest_p, IOPROTOCOL_HEADER_LENGTH + len_l + 1);
-	if (ret == 0) {
-		ret = piIoComm_recv((INT8U *) pResponse_p, REV_PI_RECV_IO_HEADER_LEN);
```

Comment:
You're removing the last occurrence of `REV_PI_RECV_IO_HEADER_LEN` from the code, so please remove the macro from piIOComm.h as well.

The macro is defined as 65530 and I did some git archeology to find out why: It turns out that prior to the serdev migration, `UartThreadProc()` contained a <a href="https://github.com/RevolutionPi/piControl/blob/master/piIOComm.c#L129">special case</a> whereby the length of the received data is taken from the packet header (instead of actually trying to receive 65530 bytes). What an egregious hack! On the serdev branch, we eliminated `UartThreadProc()` and thus the special case, so I think on the serdev branch we're currently always running into the timeout when receiving a user telegram because we're waiting for 65530 bytes. Good thing that you're fixing it now!